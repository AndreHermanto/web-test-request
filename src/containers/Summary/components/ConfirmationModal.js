import React, { Component } from 'react';
import { Modal } from 'react-bootstrap';
import { FormButton } from './../../../components/SharedStyle';
class ConfirmationModal extends Component {
  constructor(props) {
    super(props);

    this.close = this.close.bind(this);
    this.open = this.open.bind(this);
    this.state = { showModal: false };
  }

  close() {
    this.setState({ showModal: false });
  }

  open() {
    this.setState({ showModal: true });
  }

  render() {
    return (
      <span>
        <FormButton 
          onClick={this.open}
        >
          Confirm
        </FormButton> 

        <Modal show={this.state.showModal} onHide={this.close}>
          <Modal.Header closeButton>
            <Modal.Title>Confirm your request</Modal.Title>
          </Modal.Header>
          <Modal.Body>
            When you submit this test request, an invoice will be emailed to the payer if this test is privately billed.
          </Modal.Body>
          <Modal.Footer>
            <FormButton back onClick={this.close}>Cancel</FormButton>
            <FormButton type="submit" onClick={this.props.handleValidateSubmit}>
              Submit
            </FormButton>
          </Modal.Footer>
        </Modal>
      </span>
    );
  }
}

export default ConfirmationModal;
