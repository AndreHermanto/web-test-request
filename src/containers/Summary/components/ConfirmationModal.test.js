import React from 'react';
import ConfirmationModal from './ConfirmationModal';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
// import FetchMock from 'fetch-mock';

describe('ConfirmationModal', function() {  
  test('the button and modal renders correctly', function() {
    const component = renderer.create(<ConfirmationModal />).toJSON();
    expect(component).toMatchSnapshot();
  });
  
  var props = {
    handleValidateSubmit: jest.fn()
  };
  var component = TestUtils.renderIntoDocument(React.createElement(ConfirmationModal, props));
  
  test('opens the modal', function() {
    component.open();
    expect(component.state.showModal).toEqual(true);
  });
  
  test('close the modal', function() {
    component.close();
    expect(component.state.showModal).toEqual(false);
  });
  
});