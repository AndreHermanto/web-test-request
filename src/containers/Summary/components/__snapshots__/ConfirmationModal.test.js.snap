exports[`ConfirmationModal the button and modal renders correctly 1`] = `
<span>
  <button
    className="bBdbjl"
    onClick={[Function]}>
    Confirm
  </button>
</span>
`;
